client/v2/tx/factory.go

```diff
@@ -0,0 +1,555 @@
+package tx
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"math/big"
+	"strings"
+
+	"github.com/cosmos/go-bip39"
+	gogogrpc "github.com/cosmos/gogoproto/grpc"
+	"google.golang.org/protobuf/types/known/anypb"
+
+	base "cosmossdk.io/api/cosmos/base/v1beta1"
+	apicrypto "cosmossdk.io/api/cosmos/crypto/multisig/v1beta1"
+	apitxsigning "cosmossdk.io/api/cosmos/tx/signing/v1beta1"
+	apitx "cosmossdk.io/api/cosmos/tx/v1beta1"
+	"cosmossdk.io/client/v2/autocli/keyring"
+	"cosmossdk.io/client/v2/internal/account"
+	"cosmossdk.io/client/v2/internal/coins"
+	"cosmossdk.io/core/address"
+	"cosmossdk.io/core/transaction"
+	"cosmossdk.io/math"
+	"cosmossdk.io/x/tx/signing"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+	codectypes "github.com/cosmos/cosmos-sdk/codec/types"
+	"github.com/cosmos/cosmos-sdk/crypto/keys/multisig"
+	"github.com/cosmos/cosmos-sdk/crypto/keys/secp256k1"
+	cryptotypes "github.com/cosmos/cosmos-sdk/crypto/types"
+)
+
+// Factory defines a client transaction factory that facilitates generating and
+// signing an application-specific transaction.
+type Factory struct {
+	keybase          keyring.Keyring
+	cdc              codec.BinaryCodec
+	accountRetriever account.AccountRetriever
+	ac               address.Codec
+	conn             gogogrpc.ClientConn
+	txConfig         TxConfig
+	txParams         TxParameters
+
+	cachedTx TxBuilder
+}
+
+// NewFactory returns a new instance of Factory.
+func NewFactory(keybase keyring.Keyring, cdc codec.BinaryCodec, accRetriever account.AccountRetriever,
+	txConfig TxConfig, ac address.Codec, conn gogogrpc.ClientConn, parameters TxParameters,
+) (Factory, error) {
+	return Factory{
+		keybase:          keybase,
+		cdc:              cdc,
+		accountRetriever: accRetriever,
+		ac:               ac,
+		conn:             conn,
+		txConfig:         txConfig,
+		txParams:         parameters,
+	}, nil
+}
+
+// Prepare ensures the account defined by ctx.GetFromAddress() exists and
```

Comment:
Can we delete preprare and validate earlier (in the constructor, for instance)?
It looks like an not so useful extra method to call